packages/terra-navigation-prompt/docs/README.md

```diff
@@ -0,0 +1,138 @@
+# Terra Navigation Prompt
+
+The NavigationPrompt and NavigationPromptCheckpoint components allow for the detection and management of components with unsaved state.
+
+## Getting Started
+
+- Install with [npmjs](https://www.npmjs.com):
+  - `npm install terra-navigation-prompt`
+- Import the module
+  - `import NavigationPrompt, { NavigationPromptCheckpoint } from 'terra-navigation-prompt';`
+
+## Rationale
+
+A React component has complete control over its own rendering logic. It can render child components and unmount those child components as it sees fit. It follows that any individual component has no direct control over its own mounting and unmounting; it is rendered (or not) at the discretion of its parent. A component must rely on its lifecycle events to determine how it is being implemented.
+
+However, a component's children can be stateful and responsible for important user input or interactions. If the user can navigate away from that child component, it may be necessary for that child component to notify the user that their workflow is incomplete or that some transient state has not yet been persisted. If the component has no control over its own unmounting, how can it communicate this state to the overall application before the component is already being unmounted?
+
+The most straightforward solution is to lift the state up. If the state is lifted into a parent component, it can be more directly managed by the component responsible for navigation, or it could be persisted indefinitely. **This is the strategy that should be implemented by most components.** However, for very large applications or highly composable applications built out of many stateful components, lifting the state may not be feasible.
+
+Another solution is to expose callbacks from the stateful components. Whenever the stateful component's state changes, the callbacks are executed to notify the consumer of the current state. This works well at a small scale, but this becomes hard to maintain as the component tree increases in depth. One would have to chain these callbacks many levels deep to ensure every component is kept abreast of the current state, and every component, even those without these state concerns, would have to adhere to a consistent prop interface for those callbacks.
+
+The NavigationPrompt and the NavigationPromptCheckpoint were created smooth over the rough edges of the aforementioned strategies. Components that have sensitive state can render a NavigationPrompt to make their state known to the application. Components that navigate between components can use the NavigationPromptCheckpoint to check for the presence of NavigationPrompts and inform the user in a consistent way. State remains localized to the component that needs it, communication of state changes occurs automatically, and components that don't share these concerns can safely ignore all of it.
```

Comment:
- The `NavigationPrompt` and the `NavigationPromptCheckpoint`

- _rough edges of the aforementioned strategies._ should this be strategy?

- Components that have ~sensitive~ ** transient** state 

> Components that have sensitive state can render a NavigationPrompt to make their state known to the application.

its known to the application if it has implemented the `NavigationPromptCheckpoint`  or because the application has access to the context?

Reply:
> its known to the application if it has implemented the NavigationPromptCheckpoint or because the application has access to the context?

 The former. I'll rephrase.

Reply:
68704dd